engine/dbscan/src/main/java/org/opennms/alec/engine/dbscan/DBScanEngineFactory.java

```diff
@@ -46,7 +68,22 @@ public String getName() {
 
     @Override
     public AbstractClusterEngine createEngine(MetricRegistry metrics) {
-        return new DBScanEngine(metrics, epsilon, alpha, beta);
+        if(bundleContext != null) {
+            try {
+                ServiceReference<?>[] distanceMeasureRefs = bundleContext.getAllServiceReferences(DistanceMeasureFactory.class.getCanonicalName(), null);
+                for (ServiceReference<?> distanceMeasureRef : distanceMeasureRefs) {
+                    DistanceMeasureFactory factory = (DistanceMeasureFactory) bundleContext.getService(distanceMeasureRef);
+                    if (factory.getName().equalsIgnoreCase(distanceMeasureFactory)) {
+                        return new DBScanEngine(metrics, epsilon, alpha, beta, factory);
+                    }
+                }
+            } catch (InvalidSyntaxException e) {
+                throw new RuntimeException(e);
+            }
+
+            LOG.error("Wrong distance measure configuration, we'll use default");
```

Comment:
Would be good to output the value of the 'distanceMeasureFactory' specified, and the value of the default being used.